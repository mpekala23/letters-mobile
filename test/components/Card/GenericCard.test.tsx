import React from 'react';
import { Text } from 'react-native';
import { GenericCard } from '@components';
import { fireEvent, render, toJSON } from '@testing-library/react-native';

const setup = (propOverrides: Record<string, unknown> = {}) => {
  const props = {
    ...propOverrides,
  };
  return {
    ...render(<GenericCard {...props} />),
    props,
  };
};

describe('Generic Card component', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should fire onPress() on a press', () => {
    const dummy = jest.fn();
    const { getByTestId } = setup({ onPress: dummy });
    fireEvent.press(getByTestId('GenericTouch'));
    expect(dummy).toHaveBeenCalledTimes(1);
  });

  it('should render children', () => {
    const { getByText } = setup({ children: <Text>Children</Text> });
    expect(getByText('Children')).toBeDefined();
  });
});
