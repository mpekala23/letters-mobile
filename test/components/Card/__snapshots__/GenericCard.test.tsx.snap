// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic Card component should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 6,
          "marginVertical": 6,
          "padding": 15,
          "width": "100%",
        },
        Object {
          "flexDirection": "column",
          "height": 90,
          "justifyContent": "space-between",
        },
        Object {
          "elevation": 4,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2.62,
        },
        undefined,
      ]
    }
  />
</View>
`;
